### ISSUE TITLE
Rollout DiffStrategy=PlanState across providers

### ISSUE THREAD

**Issue by t0yv0** [pulumi/pulumi-terraform-bridge#866]
[Internal Design Doc](https://docs.google.com/document/d/1Zupf1LsEXKIIhtMIgsL7TryQYJdVNdzes-XlLjxvnhY/edit?usp=sharing)

The improved Diff implementation is under a feature flag DiffStrategy to make it easier to roll out across Pulumi providers gradually while mitigating any unexpected consequences. The specific risk with the new Diff implementation is that it's a sensitive method. For example, it that it might introduce unexpected replace plans to existing infrastructure on provider upgrades. The plan is:

- [ ] DiffStrategy=TryPlanState -> PlanState on 5 ecosystem providers
   - [x] hcloud
       - noting PlanState fixes https://github.com/pulumi/pulumi-hcloud/issues/170
       - but introduces https://github.com/pulumi/pulumi-hcloud/issues/175
   - [ ] aiven
       - https://github.com/pulumi/pulumi-terraform-bridge/issues/911
- [ ] DiffStrategy=TryPlanState on pulumi-gcp, pulumu-azure
  - [ ] gcp  
  - [ ] azure  
- [ ] DiffStrategy=PlanState on pulumi-gcp, pulumi-azure
  - [ ] gcp  
  - [ ] azure  
- [ ] DiffStrategy=TryPlanState on pulumi-aws
- [ ] DiffStrategy=PlanState on pulumi-aws
- [ ] pulumi-aws removes the workaround apply to make https://github.com/pulumi/pulumi-terraform-bridge/issues/791 work for the end-users, namely https://github.com/hashicorp/terraform-provider-aws/pull/25708 reintroduced into forked pulumi-aws upstream and pulumi-aws rebuilt with unmodified upstream
- [ ] DiffStrategy=PlanState by default, remove the flag

**Comment by ericrudder** [comment]
maybe explicity try gcp/azure as step 3/4 (try, plan) before aws?

i'm not really trying to add "more steps" however, *if* you think the risk
is large, this might be another baby step to consider, since aws really is
the biggest customer risk

**Comment by t0yv0** [comment]
Great idea, I'll ask again in the providers team channel to confirm. --A

**Comment by t0yv0** [comment]
Check on https://github.com/pulumi/pulumi-aws/issues/2475#issuecomment-1517892836 possibly related here.

**Comment by marco-m** [comment]
Hello, for my understanding, since @t0yv0 is unassigned, what is the plan for this issue? Currently the `hcloud` provider must be used with `pulumi up --skip-preview` to be able to deploy anything (as mentioned in the first message in this ticket). Thanks!

**Comment by t0yv0** [comment]
Hi @marco-m thanks for your feedback, I'm going to prioritize unblocking hcloud specifically given the high impact there. For the remainder of the rollout we will be reconsidering this for Q3.

**Comment by marco-m** [comment]
@t0yv0 thanks, greatly appreciated!

**Comment by t0yv0** [comment]
Awesome, great to hear. Thanks for getting back so quickly.

**Comment by t0yv0** [comment]
Noting that pulumi-aws v6.0-alpha is coming out with the flag turned on.

**Comment by t0yv0** [comment]
Closing in favor of https://github.com/pulumi/pulumi-terraform-bridge/issues/1785 - this provides a yet-closer-to-TF implementation that started rolling out to fix critical issues, and is the best way forward.
